lock-api-objects/src/main/java/com/palantir/lock/v2/TimelockService.java

```diff
@@ -110,4 +112,28 @@ default List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTr
     void tryUnlock(Set<LockToken> tokens);
 
     long currentTimeMillis();
+
+    /**
+     * Acquires a lease on a named timestamp. The lease is taken out with a new fresh timestamp.
+     * The returned timestamps are fresh timestamps obtained strictly after the lease is taken out.
+     */
+    @RestrictedApi(
+            explanation =
+                    "This method is for internal Atlas and internal library use only. Clients MUST NOT use it unless"
+                        + " given explicit approval. Mis-use can result in SEVERE DATA CORRUPTION and the API contract"
+                        + " is subject to change at any time.",
+            allowlistAnnotations = ReviewedRestrictedApiUsage.class)
+    AcquireNamedMinTimestampLeaseResult acquireNamedMinTimestampLease(String timestampName, int numFreshTimestamps);
+
+    /**
+     * Returns the smallest named timestamp for which there was an active lease at the time of the call.
+     * If there are no active leases, a fresh timestamp is obtained and returned.
+     */
+    @RestrictedApi(
+            explanation =
+                    "This method is for internal Atlas and internal library use only. Clients MUST NOT use it unless"
+                        + " given explicit approval. Mis-use can result in SEVERE DATA CORRUPTION and the API contract"
+                        + " is subject to change at any time.",
+            allowlistAnnotations = ReviewedRestrictedApiUsage.class)
+    long getMinLeasedNamedTimestamp(String timestampName);
```

Comment:
`getMinLeasedTimestampForName` or even `getMinLeasedTimestamp` given there's a parameter with name, alongside the Javadoc change of `Returns the smallest leased timestamp in the associated named collection` or similar? in both this method and the method above.